pkg/southbound/v1beta1/conn_manager.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"sync"
+
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+)
+
+// GNMIConnManager gNMI connection manager
+type GNMIConnManager interface {
+	Get(ctx context.Context, id ConnID) (*GNMIConn, error)
+	List(ctx context.Context) ([]*GNMIConn, error)
+	Watch(ctx context.Context, ch chan<- *GNMIConn) error
+	Open(conn *GNMIConn)
```

Comment:
In E2T we had to register connections in reaction to new clients connecting. In this case, since we’re connecting to the gNMI target, I think this abstraction is going to have to change a bit. For example, we may be want a `Connect` method here to create a new connection.